change/@ni-nimble-components-7b0e337d-6720-4ddb-8158-bf173072a712.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
> Alternatively we could even simplify and just while shift is being held disable text selection

What @rajsite suggested is what I have as part of my SHIFT+CLICK implementation because I ran into a similar issue where all the cells where becoming highlighted during a SHIFT+CLICK row range selection. See #1218

Reply:
Disabling text selection while Shift is down seems like a reasonable starting point, so I added that code from #1218 to this PR too. If we don't want the I-beam to show up for the table header text (or maybe just when Shift is down for that too) that could be done in a followup PR.

Reply:
It should definitely not select text on click. I agree preserving text select where we can is useful. I think we should keep the cursor (don't switch to an I-beam) and allow text selection with that.

Reply:
Updated this PR for headers to keep cursor=default (no I-beam)